### ISSUE TITLE
Feature request : new action "move and goto" (and an action selector or ordering option)

### ISSUE THREAD

**Issue by CharlesDelorme** [kewisch/quickmove-extension#181]
Using this extension saves me HOURS/week ! Thanks a lot for this work. Here is one improvement which could add more preciously saved minutes :-)

I only use two actions "move" and "goto" and I often use "move" then "goto". I do that since I also use the Thunderbird wonderful option "Place replies in the folder of the message being replied to". 

My workflow is then very often : 
1- new mail received
2- move the mail to the correct folder (and empty the inbox !)
3- move to the folder
4- reply to the mail thus the response is directly placed in the correct folder (and then having nothing to sort out of the send folder)

Actions 2 and 3 are provided by this extension and would benefit being a single action "movto" (ugly name, I must admit not being very inspired). I often mismatch my actions, choosing "move" as a second action, then moving a wrong mail to the wrong folder, then needing to move it back and so forth (and good bye preciously saved time). 

Anyway. 

My request would be to have a single action button for this double action. I have no clue of the difficulty it is to add a button (despite having to add more translations of this ugly "movto").

But I imagine maybe having too many buttons might be an UI problem. 

So a solution could be to have one or two more options : 
a- choosing which button we want to see (move/copy/goto/tag/"movto") and not see all buttons. This feature could also benefit to those not using all the actions
and/or
b- choosing the order of the buttons. This feature would also allow to change the default action, the first action being the default

These are the ideas of the #SortThisFullMailbox day…

Thank you !

**Comment by rmmbr** [comment]
I like your idea of one action for "move AND goto". I would vote for this feature ...

**Comment by kewisch** [comment]
Interesting idea. I agree adding too many more buttons could be a problem. I could imagine a toolbar-like customize feature where buttons could be re-ordered and removed, but something more simple in the options would likely be easier to implement.

**Comment by rmmbr** [comment]
I'm not a fan of additional buttons. A configurable action would be sufficient for me. This would essentially give us an invisible super feature...

**Comment by kewisch** [comment]
[quick_folder_move-3.0.0pre.zip](https://github.com/user-attachments/files/17966697/quick_folder_move-3.0.0pre.zip)

This would be quite hidden, but might be an easy way to handle it: When you press Ctrl/Cmd+Enter when moving a folder, it will follow the message and go to that folder. Users using the mouse instead of the keyboard would be out of luck.

The version above is a bit intermediate, I hit an annoying issue on keyboard handling again. If you use Cmd+Enter on mac it will not work, because it doesn't send a keyup for Enter with Cmd pressed.

What happens on Linux/Windows? If you press Ctrl+Enter, does it work?

**Comment by rmmbr** [comment]
On my System "Ctrl+Enter" doesn't work. It behaves like only "Enter".

Windows 10 Pro 22H2 Build 19045.5131. Thunderbird 128.5.0esr (64-Bit).

**Comment by rmmbr** [comment]
On my System "Ctrl+Enter" doesn't work. It behaves like only "Enter".

Windows 10 Pro 22H2 Build 19045.5131. Thunderbird 128.5.0esr (64-Bit).

**Comment by CharlesDelorme** [comment]
I'll wait for quick_folder_move-3.0.1pre.zip :-)

**Comment by rmmbr** [comment]
@CharlesDelorme : I'm looking forward to your test ... ;-)

**Comment by rmmbr** [comment]
@CharlesDelorme : I'm looking forward to your test ... ;-)

**Comment by rmmbr** [comment]
I testet with the new version. Same result.

On my System "Ctrl+Enter" doesn't work. It behaves like only "Enter".

Windows 10 Pro 22H2 Build 19045.5131. Thunderbird 128.5.1esr (64-Bit).

**Comment by kewisch** [comment]
Alright, thanks. I'll have to fire up my windows machine some time to figure it out.

**Comment by kewisch** [comment]
I tested this on my Windows 10 machine and it seems to work for me. Maybe I attached the wrong file? Here is a current build.

[quick_folder_move-3.0.0.zip](https://github.com/user-attachments/files/18059948/quick_folder_move-3.0.0.zip)

If that doesn't work, can you do Ctrl+Enter here and paste a screenshot? https://w3c.github.io/uievents/tools/key-event-viewer.html

@CharlesDelorme do you have a moment to try this out?

**Comment by rmmbr** [comment]
I testet with the new version. Same result. It doesn't work for me. 
The screenshot is attached ...
@CharlesDelorme what's on your maschine?

![Strg_Enter_12-09-2024_01](https://github.com/user-attachments/assets/8cc7be61-6f38-43e5-b04e-605e8aa5ffa8)

**Comment by rmmbr** [comment]
@kewisch Is possibly ControlLeft and ControlRight or Enter and NumpadEnter or something like that the problem?

**Comment by rmmbr** [comment]
I testet with the new version. Same result. It doesn't work for me.
The screenshot is attached ...
@CharlesDelorme what's on your machine?

![Strg_Enter_12-09-2024_01_small](https://github.com/user-attachments/assets/257c68a5-35e1-417d-a88a-1d8c41d1d158)

**Comment by CharlesDelorme** [comment]
I forgot to test ctrl+numpad enter and it is ok too

**Comment by kewisch** [comment]
> Something which might be missing is that the message (or maybe one of the messages if moving several) is not selected when jumping to the folder which might lead to the conclusion it didn't work as expected.
Ah good point, did it happen to be this @rmmbr ? I tried to add that it would select the messages, but unfortunately a "move" is actually a copy+delete, which changes the internal id. Trying to select the moved messages results in a chain of very horrible hacks with potential for a major performance impact.

I'm going to have to go with just showing the folder, not selecting the messages.

**Comment by rmmbr** [comment]
@CharlesDelorme Thanks for the precise test description.
@kewisch Sorry for the confusion I created with my testing. With Charles' description it works for me too.

My problem was the following: After selecting one message I use the comand "Shift+G" for go or "Shift+M" for move. And in the first test's I tried selecting one message and than "Ctrl+Enter" ...

@kewisch Thank you for your continued support and care of this brilliant tool.
